### DISCUSSION TITLE
How to access windmill data inside app?

### DISCUSSION THREAD

**Discussion by ghost** [windmill-labs/windmill#4217] [Questions]
I want to display latest runs of a certain flow in a select input inside a Windmill app, but the docs are not really helpful in this regard.

So I am left puzzled and have basically 2 questions:
1. How do I specify a script to run for filling the select input values?
2. How do I fetch the actual flow runs?

For a test, I have created a new AgGrid table and set the following simplistic script in a very first attempt to fetch flows:
```typescript
import { JobService } from "jsr:@windmill/windmill@1.376.1";

export async function main(offset: number, limit: number, orderBy: string, isDesc: boolean, search: string): Promise<any[]> {
  const flows = await JobService.listCompletedJobs({
    workspace: "reporting-kane",
    orderDesc: true,
    jobKinds: ["flow"],
    scriptPathStart: "f/flows/generate-report.flow",
  });

    return flows;
}
```

But then this throws:
```
TypeError: Invalid URL: '/api/w/reporting-kane/jobs/completed/list?order_desc=true&script_path_start=f%2Fflows%2Fgenerate-report.flow&job_kinds=flow'
    at getSerialization (ext:deno_url/00_url.js:98:11)
    at new URL (ext:deno_url/00_url.js:405:27)
    at new Request (ext:deno_fetch/23_request.js:329:25)
    at ext:deno_fetch/26_fetch.js:319:27
    at new Promise (<anonymous>)
    at fetch (ext:deno_fetch/26_fetch.js:315:18)
    at sendRequest (https://jsr.io/@windmill/windmill/1.376.1/src/core/request.ts:197:15)
    at https://jsr.io/@windmill/windmill/1.376.1/src/core/request.ts:316:26
    at eventLoopTick (ext:core/01_core.js:168:7)
```

Looks to me using the windmill client is not supported to be used inside apps?

**Comment by rubenfiszel** [comment]
There seem to be a bug in the jsr version of the typescript client where the client is not initialized based on context like the npm version. Will investigate but you can just use the npm version for now.

**Reply by ghost** [reply]
Thank you @rubenfiszel , that solves 2. for now.

I am still puzzled for a solution of 1. - filling the select values from such a script. Any pointers would be highly appreciated.
